backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/incremental/utils/Geometry.java

```diff
@@ -114,10 +131,126 @@ private Position getIntersection(Position lineA, Position lineB, Position segmen
+            RectangleSide side = entry.getKey();
+            Position positionOnSide = entry.getValue();
+
+            double distance = computeDistance(borderNodeCenterPosition, positionOnSide);
+            if (currentClosestDistance >= distance) {
+                boolean updateTheClosestSide = true;
+                if (currentClosestDistance == distance) {
+                    // Manage the case when borderNodeCenterPosition is outside the rectangle, near a corner of the
+                    // parent rectangle, to get the best side
+                    if (RectangleSide.NORTH.equals(side) || RectangleSide.SOUTH.equals(side)) {
+                        updateTheClosestSide = Math.abs(borderNodeCenterPosition.getY() - positionOnSide.getY()) > Math.abs(borderNodeCenterPosition.getX() - positionOnSide.getX());
+                    } else if (RectangleSide.EAST.equals(side) || RectangleSide.WEST.equals(side)) {
+                        updateTheClosestSide = Math.abs(borderNodeCenterPosition.getY() - positionOnSide.getY()) < Math.abs(borderNodeCenterPosition.getX() - positionOnSide.getX());
+                    }
+                }
+
+                if (updateTheClosestSide) {
+                    currentClosestDistance = distance;
+                    currentClosestSide = side;
+                    curentCenterPositionOnSide = positionOnSide;
+                }
+            }
+        }
+
+        // Get the new border node position according to the side and the offset
+        // 2 -
+        Position newBorderNodeCenterPosition = null;
+        if (RectangleSide.NORTH.equals(currentClosestSide)) {
+            newBorderNodeCenterPosition = Position.at(curentCenterPositionOnSide.getX() - borderNodeSize.getWidth() / 2,
+                    curentCenterPositionOnSide.getY() - borderNodeSize.getHeight() * (100 - BORDER_NODE_OFFSET_PERCENTAGE) / 100);
+        } else if (RectangleSide.SOUTH.equals(currentClosestSide)) {
+            newBorderNodeCenterPosition = Position.at(curentCenterPositionOnSide.getX() - borderNodeSize.getWidth() / 2,
+                    curentCenterPositionOnSide.getY() - borderNodeSize.getHeight() * BORDER_NODE_OFFSET_PERCENTAGE / 100);
+        } else if (RectangleSide.WEST.equals(currentClosestSide)) {
+            newBorderNodeCenterPosition = Position.at(curentCenterPositionOnSide.getX() - borderNodeSize.getWidth() * (100 - BORDER_NODE_OFFSET_PERCENTAGE) / 100,
+                    curentCenterPositionOnSide.getY() - borderNodeSize.getHeight() / 2);
+        } else if (RectangleSide.EAST.equals(currentClosestSide)) {
+            newBorderNodeCenterPosition = Position.at(curentCenterPositionOnSide.getX() - borderNodeSize.getWidth() * BORDER_NODE_OFFSET_PERCENTAGE / 100,
+                    curentCenterPositionOnSide.getY() - borderNodeSize.getHeight() / 2);
+        }
+        // get the upper right corner of the moved border node
+        @SuppressWarnings("null")
```

Comment:
I'm glad that I have added a test preventing any kind of unusual `@SuppressWarnings` to catch things like that. Fix the warning instead.